server/service/src/invoice/prescription/update/generate.rs

```diff
@@ -91,12 +93,28 @@ fn should_update_batches_total_number_of_packs(
         false
     }
 }
+// Replace datetimes that are not null with the new status_datetime
+fn replace_status_datetimes(invoice: &mut InvoiceRow, new_status_datetime: NaiveDateTime) {
+    invoice.allocated_datetime = invoice.allocated_datetime.map(|_| new_status_datetime);
+    invoice.picked_datetime = invoice.picked_datetime.map(|_| new_status_datetime);
+    invoice.verified_datetime = invoice.verified_datetime.map(|_| new_status_datetime);
+}
+
+// Handle a change to backdated_time
+fn handle_new_backdated_datetime(invoice: &mut InvoiceRow, backdated_datetime: NaiveDateTime){
+    
+    // We need to check if we are actually needing to back_date here, if the backdated_datetime is in the future, we unset the backdated_datetime as it isn't possible to future date.
+    if invoice.created_datetime < backdated_datetime {
+        // Not actually backdated!
+        let now = Utc::now().naive_utc();
+        invoice.backdated_datetime = None;
+        replace_status_datetimes(invoice, now);
```

Comment:
I think in the example I was giving this would only happen if backdated_datetime was actually set already on the invoice, although I am happy with the way you have it (would just mean picked and allocated datetime would reset anytime backdated_datetime is set > created_datetime).

Reply:
Ah true, that is kind of nice...